libgalois/src/analytics/louvain_clustering/louvain_clustering.cpp

```diff
@@ -0,0 +1,615 @@
+    katana::PropertyGraph* pg, const std::string& edge_weight_property_name,
+    const std::string& output_property_name, LouvainClusteringPlan plan) {
+  switch (pg->GetEdgeProperty(edge_weight_property_name)->type()->id()) {
+  case arrow::UInt32Type::type_id:
+    return LouvainClusteringWithWrap<uint32_t>(
+        pg, edge_weight_property_name, output_property_name, plan);
+  case arrow::Int32Type::type_id:
+    return LouvainClusteringWithWrap<int32_t>(
+        pg, edge_weight_property_name, output_property_name, plan);
+  case arrow::UInt64Type::type_id:
+    return LouvainClusteringWithWrap<uint64_t>(
+        pg, edge_weight_property_name, output_property_name, plan);
+  case arrow::Int64Type::type_id:
+    return LouvainClusteringWithWrap<int64_t>(
+        pg, edge_weight_property_name, output_property_name, plan);
+  case arrow::FloatType::type_id:
+    return LouvainClusteringWithWrap<float>(
+        pg, edge_weight_property_name, output_property_name, plan);
+  case arrow::DoubleType::type_id:
+    return LouvainClusteringWithWrap<double>(
+        pg, edge_weight_property_name, output_property_name, plan);
+  default:
+    return katana::ErrorCode::TypeError;
+  }
+}
+
+/// \cond DO_NOT_DOCUMENT
+katana::Result<void>
+katana::analytics::LouvainClusteringAssertValid(
+    [[maybe_unused]] katana::PropertyGraph* pg,
+    [[maybe_unused]] const std::string& edge_weight_property_name,
+    [[maybe_unused]] const std::string& property_name) {
+  // TODO(gill): This should have real checks. amp has no idea what to check.
+  return katana::ResultSuccess();
+}
+/// \endcond
+
+void
+katana::analytics::LouvainClusteringStatistics::Print(std::ostream& os) const {
+  os << "Total number of clusters = " << total_clusters << std::endl;
+  os << "Total number of non trivial clusters = " << total_non_trivial_clusters
+     << std::endl;
+  os << "Number of nodes in the largest cluster = " << largest_cluster_size
+     << std::endl;
+  os << "Ratio of nodes in the largest cluster = " << largest_cluster_ratio
+     << std::endl;
+  os << "Louvain modularity = " << modularity << std::endl;
+}
+
+template <typename EdgeWeightType>
+katana::Result<double>
+CalModularityWrap(
+    katana::PropertyGraph* pg, const std::string& edge_weight_property_name,
+    const std::string& property_name) {
+  using CommTy = CommunityType<EdgeWeightType>;
+  using NodeData = std::tuple<LCPreviousCommunityId>;
+  using EdgeData = std::tuple<LCEdgeWeight<EdgeWeightType>>;
+  using Graph = katana::TypedPropertyGraph<NodeData, EdgeData>;
+  using ClusterBase = katana::analytics::ClusteringImplementationBase<
+      Graph, EdgeWeightType, CommTy>;
+  ClusterBase cluster_base{};
+  auto graph_result =
+      Graph::Make(pg, {property_name}, {edge_weight_property_name});
+  if (!graph_result) {
+    return graph_result.error();
+  }
+  auto graph = graph_result.value();
+  return cluster_base.template CalModularityFinal<Graph, EdgeWeightType>(graph);
```

Comment:
The constructor of `ClusterBase` needs no arguments and is not kept around then `CalModularityFinal` can probably be a static method.